homeassistant/components/backup/manager.py

```diff
@@ -0,0 +1,181 @@
+"""Backup manager for the Backup integration."""
+from __future__ import annotations
+
+import asyncio
+from dataclasses import dataclass
+import hashlib
+import json
+from pathlib import Path
+import tarfile
+from tempfile import TemporaryDirectory
+
+from awesomeversion import AwesomeVersion
+
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import HomeAssistantError
+from homeassistant.util import json as json_util
+from homeassistant.util.dt import now
+
+from .const import EXCLUDE_FROM_BACKUP, HA_VERSION_OBJ, LOGGER
+
+
+@dataclass
+class Backup:
+    """Backup class."""
+
+    slug: str
+    name: str
+    date: str
+    path: Path
+    size: float
+
+    def as_dict(self) -> dict:
+        """Return a dict representation of this backup."""
+        return {
+            "slug": self.slug,
+            "name": self.name,
+            "date": self.date,
+            "size": self.size,
+            "path": self.path.as_posix(),
+        }
+
+
+class BackupManager:
+    """Backup manager for the Backup integration."""
+
+    def __init__(self, hass: HomeAssistant) -> None:
+        """Initialize the backup manager."""
+        self._backups: dict[str, Backup] = {}
+        self.hass = hass
+        self.backup_dir = Path(hass.config.path("backups"))
+        self.backing_up = False
+
+    @property
+    def backups(self) -> dict[str, Backup]:
+        """Return a dict of backups."""
+        return self._backups
+
+    async def load_backups(self) -> None:
+        """Load data of stored backup files."""
+        backups = {}
+
+        def _read_backup_info(backup_path: Path) -> None:
+            with tarfile.open(backup_path, "r:") as backup_file:
+                if data_file := backup_file.extractfile("./backup.json"):
+                    data = json.loads(data_file.read())
+                    backup = Backup(
+                        slug=data["slug"],
+                        name=data["name"],
+                        date=data["date"],
+                        path=backup_path,
+                        size=round(backup_path.stat().st_size / 1_048_576, 2),
+                    )
+                    backups[backup.slug] = backup
+
+        await asyncio.gather(
```

Comment:
This can starve the thread pool and does a ton of I/O in parallel. Instead, run 1 job in the executor that loops over the files.